app/controllers/obtain_annotations_with_callback_jobs_controller.rb

```diff
@@ -0,0 +1,85 @@
+class ObtainAnnotationsWithCallbackJobsController < ApplicationController
+  before_action :authenticate_root_user!, only: :new
+
+  def new
+    @project = Project.editable(current_user).find_by_name(params[:project_id])
+  end
+
+  def create
+    project = Project.editable(current_user).find_by_name(params[:project_id])
+    raise "Could not find the project: #{params[:project_id]}." unless project.present?
+    raise "Up to 10 jobs can be registered per a project. Please clean your jobs page." unless project.jobs.count < 10
+
+    # to determine the annotator
+    annotator = Annotator.find(params[:annotator])
+
+    # to determine the options
+    options = {}
+    options[:mode] = params[:mode]
+    options[:prefix] = annotator.name
+
+    # to determine the docids
+    docids = determine_docids(project, options)
+
+    # to determine the docids_filepath
+    messages = []
+    docids_filepath = determine_docids_filepath(project, docids, options, messages)
+
+    ObtainAnnotationsWithCallbackJob.perform_later(project, docids_filepath, annotator, options)
+
+    project.update({annotator_id:annotator.id}) if annotator.persisted?
+
+    messages << "The task 'Obtain annotations was created."
+    message = messages.join("\n")
+
+    respond_to do |format|
+      format.html {redirect_back fallback_location: root_path, notice: message}
+      format.json {}
+    end
+  rescue => e
+    respond_to do |format|
+      format.html {redirect_back fallback_location: root_path, notice: e.message}
+      format.json {render status: :service_unavailable}
+    end
+  end
+
+  private
+
+  def determine_docids(project, options)
+    if options[:mode] == 'fill'
+      options[:mode] = 'add'
+      ProjectDoc.where(project_id:project.id, annotations_updated_at:nil).pluck(:doc_id)
```

Comment:
```suggestion
      project.project_docs.where(annotations_updated_at:nil).pluck(:doc_id)
```

と書けそうです。

さらに`where(annotations_updated_at:nil)`をProjectDocクラスのscopeにして`project.project_docs.without_annotation.pluck(:doc_id)`と書けると良いです。

しかし、これはProjectのプロパティとしてとれると良さそうです。
ということはProjectに次のようなアソシエーションを定義すると

```ruby
  has_many :project_docs, dependent: :destroy
  has_many :project_docs_without_annotation, -> { where(annotations_updated_at: nil) }, class_name: 'ProjectDoc'
  has_many :docs, through: :project_docs
  has_many :docs_without_annotation, through: :project_docs_without_annotation, source: :doc
```
次のように書けそうです。

```ruby
project.docs_without_annotation.pluck(:id)
```

Reply:
修正しました！
https://github.com/pubannotation/pubannotation/pull/117/commits/dd668607daff89f5a267b278f0992ea0e97d85e8